### ISSUE TITLE
Enable Certificate authentication option in API template

### ISSUE THREAD

**Issue by mkArtakMSFT** [aspnet/AspNetCore#10336]
This is listed in the Templating epic with questionmark, so not sure what exactly needs to happen here or how important it is to get it done in 3.0.

**Comment by mkArtakMSFT** [comment]
This depends on https://github.com/aspnet/AspNetCore/issues/4663

**Comment by blowdart** [comment]
Given the desire to drive templates to be smaller, this is probably a rare enough want that it doesn't need template integration, but documentation will suffice.

**Comment by anurse** [comment]
Agreed. I don't think we should overload the template with samples. We just finished scaling it back :)
